lib/logwriter.c

```diff
@@ -247,9 +244,9 @@ log_writer_io_flush_output(gpointer s)
   main_loop_assert_main_thread();
 
   log_writer_stop_watches(self);
-  log_pipe_ref(&self->super);
   if ((self->options->options & LWO_THREADED))
     {
+      log_pipe_ref(&self->super);
       main_loop_io_worker_job_submit(&self->io_job);
     }
```

Comment:
I think this could be made a bit more explicit that self->io_job->user_data equals to self. What about this
```
self->io_job->user_data = log_pipe_ref(&self->super);
```
and the same on the logreader side.